client/src/components/common/SideNav.jsx

```diff
@@ -10,9 +11,11 @@ import PropTypes from 'prop-types';
 class SideNav extends Component {
   constructor(props) {
     super(props);
-    this.state = {
-      search: '',
-    };
+    this.handleFilter = this.handleFilter.bind(this);
```

Comment:
In other not to bind function in the constructor like this, you can define the function as arrow function which auto binds functions. For instance, you can redefine the `handleFilter` function as follows:
```javacript
handleFilter = (event) => {
  this.props.filterIdeas(event.target.id);
}
```
Install `transform-class-properties` package and include the following in `.babelrc`.
{
"plugins": [
    "transform-class-properties"
 ]
}